lib/srv/app/aws/handler_test.go

```diff
@@ -296,6 +319,23 @@ func TestAWSSignerHandler(t *testing.T) {
 				require.NoError,
 			},
 		},
+		{
+			name: "Lambda access",
```

Comment:
Does this test anything as the limit is checked outside of AWS handler?

Reply:
this test only ensures that the handler does not enforce a limit

Reply:
done. I also noticed that the status code change from 429 to 500 when the error was encountered, and io.LimitedReader only returns EOF after the limit is reached, unlike what our `ReadAtMost` util which returns a limit exceeded error. So I created a couple wrapper types to ensure the old behavior is preserved in ddc05a44338ef04686f8570fcdce525d41455112

Reply:
Why do we use `MaxBytesReader` in some places but `LimitReader` in other places? Is `MaxBytesReader` an optimization? We do have `formatForwardResponseError` to set the HTTP code but i assume we want to add a new type in trace eventually.

Reply:
`MaxBytesReader` take an http.ResponseWriter and does some early close stuff if the writer supports it - basically an optimization. Used when you have a proper http.Server and a ServeHTTP handler where the response writer is available.

Elsewhere, just use the LimitedReader and errors go through a function like `formatForwardResponseError` that does some custom error code things and writes an error response.

Since the status code was already 429 (too many request) for these errors, I don't think it's in scope within this PR to make it return 413 (request too big). I also think adding a trace type or a util for converting http errors will make sense, but doing it in a separate PR would be better.